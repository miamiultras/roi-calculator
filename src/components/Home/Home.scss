@import '../../styles/typography';

.title-container {
  z-index: 10;
  opacity: 1;
  width: 70%;
  color: #fff;
  h1 {
    font-family: $global-thin-font-family;
    color: #fff;
    font-size: 48px;
    font-weight: 400;
    margin: 20px 0 10px 0;
  }
  .description {
    font-family: $global-light-font-family;
    color: #fff;
    font-size: 24px;
    line-height: 1.6;
  }
  button {
    font-family: $global-regular-font-family;
    font-size: 20px;
    color: #fff;
    width: auto;
    margin: 60px 0px;
    padding: 15px 35px;
    background: linear-gradient(-21deg, rgb(97, 186, 166), rgb(99, 189, 116));
    border: 1px rgb(97, 186, 166);
    border-radius: 32px;
    cursor: pointer;
  }
}
