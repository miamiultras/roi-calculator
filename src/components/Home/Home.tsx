import React from 'react';
import { Link } from 'react-router-dom';

import './Home.scss';

const Home = () => (
  <div className='home-container'>
    <div className='title'>
      <h1>Live PPV Event ROI Calculator</h1>
    </div>
    <div className='description'>
      Ever wondered <strong>how much money you can generate </strong>
      with the Live Pay-Per-View model? Based on our experience in
      <strong> managing over 10,000 PPV</strong> events per year,
      we created this unique ROI calculator for you.
    </div>
    <Link to='/calculator'>
      <button>Give it a try!</button>
    </Link>
  </div>
);

export default Home;
