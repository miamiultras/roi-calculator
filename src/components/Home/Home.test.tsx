import * as React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';

function setup() {
  const props = {};
  const wrapper = shallow(<Home />);
  return {
    wrapper,
    props,
  };
}

describe('Home component', () => {
  it('should match snapshot if no change was intended', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
