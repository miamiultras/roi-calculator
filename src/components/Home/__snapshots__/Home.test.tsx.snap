// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match snapshot if no change was intended 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Live PPV Event ROI Calculator
        </h1>,
        <div
          className="description"
        >
          Ever wondered 
          <strong>
            how much money you can generate 
          </strong>
          with the Live Pay-Per-View model? Based on our experience in
          <strong>
             managing over 10,000 PPV
          </strong>
           events per year, we created this unique ROI calculator for you.
        </div>,
        <Link
          replace={false}
          to="/calculator"
        >
          <button>
            Give it a try!
          </button>
        </Link>,
      ],
      "className": "home-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Live PPV Event ROI Calculator",
        },
        "ref": null,
        "rendered": "Live PPV Event ROI Calculator",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Ever wondered ",
            <strong>
              how much money you can generate 
            </strong>,
            "with the Live Pay-Per-View model? Based on our experience in",
            <strong>
               managing over 10,000 PPV
            </strong>,
            " events per year, we created this unique ROI calculator for you.",
          ],
          "className": "description",
        },
        "ref": null,
        "rendered": Array [
          "Ever wondered ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "how much money you can generate ",
            },
            "ref": null,
            "rendered": "how much money you can generate ",
            "type": "strong",
          },
          "with the Live Pay-Per-View model? Based on our experience in",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " managing over 10,000 PPV",
            },
            "ref": null,
            "rendered": " managing over 10,000 PPV",
            "type": "strong",
          },
          " events per year, we created this unique ROI calculator for you.",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button>
            Give it a try!
          </button>,
          "replace": false,
          "to": "/calculator",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Give it a try!",
          },
          "ref": null,
          "rendered": "Give it a try!",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Live PPV Event ROI Calculator
          </h1>,
          <div
            className="description"
          >
            Ever wondered 
            <strong>
              how much money you can generate 
            </strong>
            with the Live Pay-Per-View model? Based on our experience in
            <strong>
               managing over 10,000 PPV
            </strong>
             events per year, we created this unique ROI calculator for you.
          </div>,
          <Link
            replace={false}
            to="/calculator"
          >
            <button>
              Give it a try!
            </button>
          </Link>,
        ],
        "className": "home-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Live PPV Event ROI Calculator",
          },
          "ref": null,
          "rendered": "Live PPV Event ROI Calculator",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Ever wondered ",
              <strong>
                how much money you can generate 
              </strong>,
              "with the Live Pay-Per-View model? Based on our experience in",
              <strong>
                 managing over 10,000 PPV
              </strong>,
              " events per year, we created this unique ROI calculator for you.",
            ],
            "className": "description",
          },
          "ref": null,
          "rendered": Array [
            "Ever wondered ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "how much money you can generate ",
              },
              "ref": null,
              "rendered": "how much money you can generate ",
              "type": "strong",
            },
            "with the Live Pay-Per-View model? Based on our experience in",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " managing over 10,000 PPV",
              },
              "ref": null,
              "rendered": " managing over 10,000 PPV",
              "type": "strong",
            },
            " events per year, we created this unique ROI calculator for you.",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
              Give it a try!
            </button>,
            "replace": false,
            "to": "/calculator",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Give it a try!",
            },
            "ref": null,
            "rendered": "Give it a try!",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
